src/com/owncloud/android/lib/common/accounts/AccountUtils.java

```diff
@@ -107,6 +109,29 @@ public static String getBaseUrlForAccount(Context context, Account account)
 
 
     /**
+     * Extracts server url for a file
+     * @param context
+     * @param account
+     * @param fileId remote file id
+     * @return url server for file or null on failure
+     */
+    public static String getUrlForFile (Context context, Account account, String fileId)
```

Comment:
I would search for another class to place this method, better if it's not in the OC-library. The method implementation might call ```AccountUtils#getBaseUrlForAccount(...)```  if there is no other better way to get the base URL, but responsibilities of ```AccountUtils```  should not include know specific things about individual files.